### ISSUE TITLE
Allow `min_periods`  in `rolling_` functions to take dynamic temporal size

### ISSUE THREAD

**Issue by dylan-lee94** [pola-rs/polars#13566]
### Description

I'm using polars.Expr.rolling_sum for stock return calculations, in particular for computing 1 month trailing stock returns. Here it would be great if `min_periods` ensures that the 1 month trailing return is `Null `when there is not enough data available for the entire window size.

Let's create some dummy stock returns 

```
from datetime import date
import polars as pl
import numpy as np

np.random.seed(42)

df = pl.date_range(date(2022, 1, 1), date(2022, 3, 1), interval="1d", eager=True).alias("CALENDAR_DATE").to_frame()
df = df.with_columns(pl.Series(name="VALUE", values=np.random.randint(90, high=110, size=df.height, dtype=int)))
df = df.with_columns(pl.col("VALUE").pct_change().alias("DAILY_RETURN")).drop_nulls()
```

In the current implementation, `min_periods` is set to 1, if window_size is a dynamic temporal size. 
```
df1 = df.sort('CALENDAR_DATE')\
        .with_columns(((1+pl.col('DAILY_RETURN')).log().rolling_sum(by='CAL
ENDAR_DATE', window_size='1mo').exp()-1)\
        .alias('1_MONTH_TRAILING_RETURN')
        )
```
[df1.csv](https://github.com/pola-rs/polars/files/13875297/df1.csv)

It would be great to allow `min_periods` to take a dynamic temporal size, similarly to:

```
df2 = df.sort('CALENDAR_DATE')\
        .with_columns(((1+pl.col('DAILY_RETURN')).log().rolling_sum(by='CALENDAR_DATE', window_size='1mo', min_periods='1mo').exp()-1)\
        .alias('1_MONTH_TRAILING_RETURN')
        )
```
[df2.csv](https://github.com/pola-rs/polars/files/13875298/df2.csv)

I am not entirely sure, but this might already be addressed by #12049.

**Comment by orlp** [comment]
Do I understand you correctly that you'd want a `mininum_span` parameter such that when the samples in the window span a period in length less than this quantity null is returned?

If so we could consider adding that after the rework into `rolling` and `rolling_interval`, but it would be blocking work right now.

**Comment by dylan-lee94** [comment]
Yes exactly. In the above described example, with a dynamic `mininum_span` argument of `1mo `I would expect the function to return values once the window spans a period length from 2022-02-02 to 2022-03-02 (depending on `closed` argument).
